src/agents/combined_context_graph.py

```diff
@@ -1,55 +1,24 @@
 import functools
-import operator
-
-from typing import TypedDict, Annotated, Optional
-from pydantic import Field
-from langchain_core.messages import BaseMessage
-
 from langgraph.graph import END, StateGraph, START
-
-
 from langfuse.callback import CallbackHandler
 from langchain.callbacks.manager import CallbackManager
 import config
 from agents.utils import *
+from models.agent_state import AgentState
 from langgraph.checkpoint.memory import MemorySaver
-from chains.qvkg_chain import QVKGChain
+from chains import QVKGChain, UserIntentChain
 import config as app_config
-from guardrails.input_guard import InputGuard
```

Comment:
also moved the guard rail node construction into (https://github.com/helxplatform/Koios/blob/user-intent-integeratio/src/agents/utils.py#L71) cause, its going to be a shared thing , no need to re-write it every time.